feature/experimental/tunnel/otg_tests/tunnel_interface_based_ipv6_gre_encapsulation_test/tunnel_interface_based_ipv6_gre_encapsulation_test.go

```diff
@@ -0,0 +1,467 @@
+			t.Logf("No Traffic Loss Test Passed!!")
+		}
+	} else {
+		if lossPct < 100-uint64(tolerance) {
+			t.Errorf("Traffic is expected to fail but flow :%s  got %v, want 100%% Failed.", flowName, lossPct)
+		} else {
+			t.Logf("Traffic Loss Test Passed!!")
+		}
+	}
+}
+func sendTraffic(t *testing.T, ate *ondatra.ATEDevice) {
+	otg := ate.OTG()
+	t.Logf("Starting traffic")
+	otg.StartTraffic(t)
+	time.Sleep(time.Duration(trafficDuration) * time.Second)
+	t.Logf("Stop traffic")
+	otg.StopTraffic(t)
+}
+func configureOtgPorts(top gosnappi.Config, port *ondatra.Port, name string, mac string, ipv6Address string, ipv6Gateway string, ipv6Mask uint8) {
+
+	top.Ports().Add().SetName(port.ID())
+	//port1
+	iDutDev := top.Devices().Add().SetName(name)
+	iDutEth := iDutDev.Ethernets().Add().SetName(name + ".Eth").SetMac(mac)
+	iDutEth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.PORT_NAME).SetPortName(port.ID())
+	iDutIpv6 := iDutEth.Ipv6Addresses().Add().SetName(name + ".IPv6")
+	iDutIpv6.SetAddress(ipv6Address).SetGateway(ipv6Gateway).SetPrefix(uint32(ipv6Mask))
+}
+func configureTrafficFlowsToEncasulation(t *testing.T, top gosnappi.Config, port1 *ondatra.Port, port2 *ondatra.Port, port3 *ondatra.Port, peer *attrs.Attributes, destMac string) {
+	t.Logf("configure IPv6 flow from %s ", port1.Name())
+	// Set config flow
+	flow1ipv6 := top.Flows().Add().SetName("IPv6")
+	flow1ipv6.Metrics().SetEnable(true)
+	//TxRx ports
+	flow1ipv6.TxRx().Port().SetTxName(port1.ID()).SetRxNames([]string{port2.ID(), port3.ID()})
+	// Flow settings.
+	flow1ipv6.Size().SetFixed(512)
+	flow1ipv6.Rate().SetPps(trafficRatePps)
+	flow1ipv6.Duration().SetChoice("continuous")
+	// Ethernet header
+	f1e2 := flow1ipv6.Packet().Add().Ethernet()
+	f1e2.Src().SetValue(peer.MAC)
+	f1e2.Dst().SetValue(destMac)
+	// IPv6 header
+	f1v6 := flow1ipv6.Packet().Add().Ipv6()
+	// V6 NextHeader
+	f1v6.NextHeader().SetChoice("value").SetValue(6)
+	// V6 source
+	f1v6.Src().Increment().SetStart(peer.IPv6).SetCount(200)
+	// V6 destination
+	f1v6.Dst().Increment().SetStart(encapInnerDesIpv6Network).SetCount(200)
+	// Add L4 protocol
+	flow1ipv6.Packet().Add().Tcp()
+	// Increment Source port
+	flow1ipv6.Packet().Add().Tcp().SrcPort().Increment().SetStart(1000).SetCount(2000)
+	// Increment destination port
+	flow1ipv6.Packet().Add().Tcp().DstPort().Increment().SetStart(4000).SetCount(2000)
+}
+func fetchNetworkAddress(t *testing.T, address string, mask int) (string, string) {
+	addr := net.ParseIP(address)
+	var network net.IP
+	// This mask corresponds to a /128 subnet for IPv6.
+	ipv6Mask := net.CIDRMask(mask, 128)
+	network = addr.Mask(ipv6Mask)
+	networkWithMask := network.String() + "/" + strconv.Itoa(mask)
+	networkAlone := network.String()
+	//t.Logf("Network address : %s", networkWithMask)
+	return networkAlone, networkWithMask
+}
+func incrementAddress(t *testing.T, address string, i int, part string) string {
+	addr := net.ParseIP(address)
+	var oct int
+
+	if part == "network" {
```

Comment:
use a switch statement. Can part be anything other than "network" or "host". If not, fail fatally in that case

Reply:
moved to a switch statement